lib/trivial.nix

```
@@ -448,6 +448,39 @@ rec {
   isFunction = f: builtins.isFunction f ||
     (f ? __functor && isFunction (f.__functor f));
 
+  /* Copy the function arguments (__functionArgs) of the first function
+     onto the second function.
+
+     Type:
+       copyFunctionArgs :: ({ <a_i> } -> b) -> (a -> c) -> (({ <a_i> } -> c) & AttrSet)
+
+     Example:
+       addab = {a, b}: a + b
+       addab { a = 2; b = 4; }
+       => 6
+       lib.functionArgs addab
+       => { a = false; b = false; }
+       addab1 = attrs: addab attrs + 1
+       addab1 { a = 2; b = 4; }
+       => 7
+       lib.functionArgs addab1
+       => { }
+       addab1' = lib.copyFunctionArgs addab addab1
+       addab1' { a = 2; b = 4; }
+       => 7
+       lib.functionArgs addab1'
+       => { a = false; b = false; }
+   */
+  copyFunctionArgs =
```

```suggestion
  mirrorFunctionArgs =
```

I like this wording more because "copy" feels a bit too immutable.

---

Cool! I like the new name.